application/common/models/MfaBackupcode.php

```diff
@@ -31,10 +31,26 @@ public function rules(): array
      */
     public static function validateAndRemove(int $mfaId, $code): bool
     {
+        $intCode = intval($code);
+        $codeStartsWithZero = (! empty($code) && $code[0] === '0');
```

Comment:
The code...
`(! empty($code) && $code[0] === '0')`
... is complicated enough, I have to look closely and think for a bit to decide whether the `!` applies to the `empty()` or to the result of the `&&`:
`((! empty($code)) && $code[0] === '0')`
vs.
`(! (empty($code) && $code[0] === '0'))`

What would you think of simplifying it to something like this?
`substr($code, 0, 1) === '0'`
That works even if an empty string was given, though it's possible I overlooked some different shortcoming.

Thoughts?

Reply:
Fixed.